### DISCUSSION TITLE
Is it possible to get only the key for an attachment stored in s3?

### DISCUSSION THREAD

**Discussion by Bandes** [shrinerb/shrine#674] [Q&A]
Hi - I know that if I have an attachment stored in s3 I can do something like `record.file_url` to get a url to the file - but I would like to get just the key to the file because I have some non-shrine file operations I need to do with the attachment. I know that I could do some regex or `split` nonsense on the url to extract the key, but I am hoping that there might be a shrine attacher method that I don't know about which could just return the key for me directly?

Maybe I just need `"#{record.file_data['storage']}/#{record.file_data['id']}"`?

**Comment by jrochkind** [comment]
Hi, what do you mean by the "key" to the file?  Can you give an example?  Woudln't non-shrine file operations still need the URL to get access to the file on S3?  Or do you want a `s3://` uri identifier, is that the "key"? 

Or are you maybe looking for `record.file_url(public: true)`?

**Reply by Bandes** [reply]
No, I want to access these files through the S3 SDK directly, and in order to do so I need the identifier that S3 uses as the `key` - this is part of the `file_url` but not the whole thing. I believe that the key is the `storage` plus the `id` so I think I can just construct that, but I was hoping there might be a simpler way.
